### PR TITLE
Feature/i18n support

### PR DESCRIPTION
**Pull request by emclaug2**
#### Changes proposed in this Pull Request:
-  Add spanish, french, chinese, english translations. 
- Update documentation.

Custom languages can be supported like this:
```
 this.pxbAuthConfig.customTranslations = pxbCustomTranslations;
```

Hosted here: https://pxblue-temp.web.app/

### CODE REVIEW

**Comment by jeffvg** [comment]
anything special here to run? My local branch isn't switching to different languages but I see the deployed version is working

**Comment by emclaug2** [comment]
> anything special here to run? My local branch isn't switching to different languages but I see the deployed version is working

Are you going into the debug mode and switching languages from the login screen?

**Comment by jeffvg** [comment]
just here

image

**Comment by jeffvg** [comment]
will we be setting language based on browser settings like React or is that future work

**Comment by emclaug2** [comment]
> PxbAuthTranslations

For this example to work with a custom file, you need a custom translation file to import.  You can do this by copy/pasting the src/translations/english.ts file into example/src/app.  Maybe call it `customTranslations.ts`. 

Change the export to `pxbCustomTranslations ` and add:
`import { PxbAuthTranslations } from '@pxblue/angular-auth-workflow';`

In your app.component:
`import { pxbCustomTranslations } from './customTranslations';`

```
_configurePxbAuthModule(): void {
...
this.pxbAuthConfig.customTranslations = pxbCustomTranslations;
...
}
```

**Comment by emclaug2** [comment]
> I'm unable to get this running using 'this.pxbAuthConfig.customTranslations'
> Currently the app.component has:
> import {PxbAuthTranslations} from "../../../dist";
> and this.pxbAuthConfig.languageCode = 'EN'
> 
> setting language code not working either...not sure how to run this branch
> 
> languageCode** (optional): _`'EN' | 'FR' | 'ES' | 'ZH'`

The language codes are causing an issue; I'll fix that right now.

**Comment by emclaug2** [comment]
@jeffvg  The language codes were not being used correctly; this should work now.

**Review by daileytj — APPROVED** [review]
Looks good to me!
Screen Shot 2021-05-12 at 3 48 56 PM

**Review by jeffvg — APPROVED** [review]

File: login-workflow/docs/services.md [hunk 1]
```diff
@@ -51,15 +51,18 @@ constructor(pxbAuthConfig: PxbAuthConfig) {
     -   Custom email regex requirements.
     -   Default: `undefined`
 -   **customPasswordRequirements** (optional): _`PasswordRequirement[]`_
-    -   Array of custom password requirements.
+    -   Array of custom password requirements. 
     -   Default: `[]`
+-   **customTranslations** (optional): _`PxbAuthTranslations`_
```
Comment (joebochill) [inline login-workflow/docs/services.md:56]: Can we add some documentation around this? What happens if a user supplies a custom translation file with missing fields? Should we link them (or point them) to information about what strings they will have to include?

File: login-workflow/src/translations/spanish.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
+/* eslint-disable @typescript-eslint/naming-convention */
```
Comment (joebochill) [inline login-workflow/src/translations/spanish.ts:1]: Mariana sent me some tweaks for the Spanish translations. I haven't updated them for the shared auth package yet, but we'll probably need to make them here too (we can do a follow-up for that, just letting you know).

**Review by joebochill — COMMENTED** [review]
Changelog, version?

File: login-workflow/docs/services.md (hunk 1 above)
Comment (emclaug2) [reply login-workflow/docs/services.md:56]: We have a type `PxbAuthTranslations` for them to use; as long as they use that type then they can't accidently forget fields.

Comment (emclaug2) [reply login-workflow/docs/services.md:56]: I can add a blurb saying to look at the other translation files for reference.

File: login-workflow/src/translations/spanish.ts (hunk 2 above)
Comment (emclaug2) [reply login-workflow/src/translations/spanish.ts:1]: Yeah that sounds good to me; our translation string names aren't a one-to-one match, so it's a little tedious when trying to find the matching string between the two packages.

**Pull request merged**